torch/_library/fake_class_registry.py

```diff
@@ -99,9 +99,18 @@ def _check_valid_flat_script_obj(flat_x):
             )
 
 
-def to_fake_obj(fake_mode, x: torch.ScriptObject) -> FakeScriptObject:
+def tracing_with_real(x: torch.ScriptObject) -> bool:
+    return hasattr(x, "safe_to_trace_with_real_obj") and x.safe_to_trace_with_real_obj()
```

Comment:
API bikeshedding:

What if we just had a "tracing_mode" flag that was either "real" or "fake"?